bin/src/_main.rs

```diff
@@ -162,22 +163,48 @@ pub async fn _main(
                 if config.log.use_k8s_enrichment == K8sTrackingConf::Always
                     && std::env::var_os("KUBERNETES_SERVICE_HOST").is_some()
                 {
+                    let middleware = Arc::new(std::sync::Mutex::new(K8sMetadata::default()));
+                    executor.register_arc(middleware.clone());
+
                     let node_name = std::env::var("NODE_NAME").ok();
-                    match K8sMetadata::new(k8s_client.clone(), node_name.as_deref()).await {
-                        Ok((driver, v)) => {
-                            tokio::spawn(driver);
-                            executor.register(v);
-                            info!("Registered k8s metadata middleware");
-                        }
-                        Err(e) => {
-                            let message = format!(
-                                "The agent could not access k8s api after several attempts: {}",
-                                e
-                            );
-                            error!("{}", message);
-                            panic!("{}", message);
+                    let k8s_client_clone = k8s_client.clone();
+                    tokio::spawn(async move {
```

Comment:
client is reused? connection is not recreated?
and we reuse the storage?

Reply:
Ah I didn't think about the client, it makes sense to recreate it every time in case our issue is arising there. The reflector store is dropped with the thread each refresh; so a new store and watcher is created each time.

Reply:
I assume while we restart watcher we will not lose notifications. could you check API spec that we do get some backlog of notifications after new connect?